LumoraGodot/Source/Godot/Audio/Audio.cs

```diff
@@ -0,0 +1,205 @@
+        return AudioServer.GetBusVolumeDb(_busId);
+    }
+    public float GetVolumeNormalized()
+    {
+        if (!_isValid)
+            throw new Exception("This Bus Refrence is no longer valid");
+        return AudioServer.GetBusVolumeLinear(_busId);
+    }
+    public string Name
+    {
+        get
+        {
+            if (!_isValid)
+                throw new Exception("This Bus Refrence is no longer valid");
+            return AudioServer.GetBusName(_busId);
+        }
+        set
+        {
+            if (!_isValid)
+                throw new Exception("This Bus Refrence is no longer valid");
+            AudioServer.SetBusName(_busId, value);
+        }
+    }
+    public List<IAudioEffect> Effects
+    {
+        get { throw new NotImplementedException(); }
+    }
+    public IAudioBus Target
+    {
+        get
+        {
+            if (!_isValid)
+                throw new Exception("This Bus Refrence is no longer valid");
+            return AudioMixer.GetMixer().GetAudioBusByName(AudioServer.GetBusSend(_busId));
+        }
+        set
+        {
+            if (!_isValid)
+                throw new Exception("This Bus Refrence is no longer valid");
+            AudioServer.SetBusSend(_busId, value.Name);
+        }
+    }
+    internal int BusId
+    {
+        get => _busId;
+    }
+    internal AudioBus(int busid)
+    {
+        Lumora.Core.Logging.Logger.Log("creating bus");
+        _isValid = true;
+        _busId = busid;
+    }
+    public float Channels { get => 2; }
+}
+public partial class AudioMixer : IAudioMixer
+{
+
+    public string[] GetAvailableAudioEffects()
+    {
+        throw new NotImplementedException();
+    }
+    public static AudioMixer GetMixer() => _instance;
+    private static readonly AudioMixer _instance = new AudioMixer();
+    private readonly object _writeLock = new object();
+    private readonly Dictionary<string, AudioBus> buses = new();
+    private AudioMixer() { ForceSync(); }
+    public void ForceSync()
+    {
+        foreach (var kv in buses)
+        {
+            kv.Value.Invalidate();
+        }
+        lock (_writeLock)
+        {
+
+            Lumora.Core.Logging.Logger.Log($"[Audio][ForceSync] There are {AudioServer.BusCount} buses in the engine");
+            buses.Clear();
+            for (int i = 0; i < AudioServer.BusCount; i++)
+            {
+                string name = AudioServer.GetBusName(i);
+                Lumora.Core.Logging.Logger.Log($"[Audio][ForceSync] Adding bus {name}");
+                var bus = new AudioBus(i);
+                Lumora.Core.Logging.Logger.Log($"[Audio][ForceSync] the bus exists and its name is {bus.Name}");
+                buses.Add(name, bus);
+            }
+        }
+    }
+    public IAudioBus GetAudioBusByName(string name)
+    {
+        if (buses.TryGetValue(name, out var b))
+            return b;
+        throw new Exception("I told you so");
```

Comment:
why throw a excpetion like this... in prod code 
let the user decide what to do with this excpetion thrown